receiver/azureblobreceiver/receiver.go

```diff
@@ -18,22 +18,108 @@ import (
 	"context"
 
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/obsreport"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
 )
 
+type LogsDataConsumer interface {
+	ConsumeLogsJSON(ctx context.Context, json []byte) error
+	SetNextLogsConsumer(nextLogsConsumer consumer.Logs)
+}
+
+type TracesDataConsumer interface {
+	ConsumeTracesJSON(ctx context.Context, json []byte) error
+	SetNextTracesConsumer(nextracesConsumer consumer.Traces)
+}
+
 type blobReceiver struct {
+	blobEventHandler   BlobEventHandler
+	logger             *zap.Logger
+	logsUnmarshaler    plog.Unmarshaler
+	tracesUnmarshaler  ptrace.Unmarshaler
+	nextLogsConsumer   consumer.Logs
+	nextTracesConsumer consumer.Traces
+	obsrecv            *obsreport.Receiver
 }
 
 func (b *blobReceiver) Start(ctx context.Context, host component.Host) error {
-	return nil
+
+	b.blobEventHandler.SetLogsDataConsumer(b)
+	b.blobEventHandler.SetTracesDataConsumer(b)
+
+	err := b.blobEventHandler.Run(ctx)
+
+	return err
 }
 
 func (b *blobReceiver) Shutdown(ctx context.Context) error {
+	b.blobEventHandler.Close(ctx)
+
 	return nil
```

Comment:
You should not ignore the error here:
```suggestion
	return b.blobEventHandler.Close(ctx)
```